posts/2021/09/array-libraries-interoperability.md

```diff
@@ -0,0 +1,537 @@
+started with adding an alias `torch.concat` for `torch.cat` which is
+how concatenation is defined in the [array api spec](https://data-apis.org/array-api/latest/API_specification/manipulation_functions.html?highlight=concat#concat-arrays-axis-0)
+and ended with a bunch of other fixes related to concatenation in PyTorch.
+Later, I worked on improving compatibility for Array API to the `torch.linalg`
+module. See [pytorch/pytorch#63285](https://github.com/pytorch/pytorch/pull/63285)
+and [pytorch/pytorch#63227](https://github.com/pytorch/pytorch/pull/63227).
+
+You can track the progress on PyTorch's github repo with the label
+["module: python array api"](https://github.com/pytorch/pytorch/labels/module%3A%20python%20array%20api) to check out other interesting developments.
+
+ 
+## uarray
+
+uarray is a backend system for Python that allows you to separately define
+an API, along with backends that contain separate implementations of that API.
+
+I've been working on adding uarray backend support to more SciPy modules. 
+
+<p align="center">
+	<a href="https://github.com/scipy/scipy/issues/14353">
+	<img alt="SciPy: uarray compatibility tracker" src="/images/2021/09/uarray_compatibility_tracker.png">
+    <i>SciPy: uarray compatibility tracker</i>
+    </a>
+</p>
+
+The uarray backend compatibility tracker issue linked above sums up
+the plan and current state of uarray in SciPy.
+
+Precisely I've been working on adding uarray support in the
+[`scipy.ndimage`](https://github.com/scipy/scipy/tree/master/scipy/ndimage) module
+for the last couple of months.
+See [scipy/scipy#14356](https://github.com/scipy/scipy/pull/14356) for more
+details on the discussions.
+
+With the `ndimage` supporting `uarray` backend soon, one would be able to
+achieve the following in the future:
+
+```python
+# SciPy ndimage with CuPy array
+from scipy import ndimage
+import cupy as cp
+ 
+with scipy.ndimage.set_backend('cupy'):
+    y_cupy = ndimage.correlate1d(cp.arange(10),
+                                 cp.array([1, 2.5]))
+```
+
+The [work](https://github.com/scipy/scipy/pull/14356) on adding `uarray`
+backend in `ndimage` is slightly complicated and
+involved a few other maintainace fixes in SciPy. In case you are interested,
+I'll leave a list of some notable SciPy contributions below which are directly
+or indirectly connected to uarray.
+
+* [scipy/scipy#14275](https://github.com/scipy/scipy/pull/14275)
+* [scipy/scipy#14359](https://github.com/scipy/scipy/pull/14359)
+* [scipy/scipy#14447](https://github.com/scipy/scipy/pull/14447)
+* [scipy/scipy#14474](https://github.com/scipy/scipy/pull/14474)
+* [scipy/scipy#14266](https://github.com/scipy/scipy/issues/14266)
+
+---
+
+
+## All ☀️ & 🌈?
+
+As they say, nothing is perfect on the human stage, both uarray and
+Array API also have their limitations.
+
+\*\*\*_Wears interoperability hat again_\*\*\*
+
+Let's highlight some Pros & Cons for these two protocols.
```

Comment:
This actually explains a lot, like where uarray comes in. Regarding my other comment about making the transition smoother: you can do that early on in the blog post, and then forward-link to this section for a more detailed explanation.